spec/system/nested_documentable_spec.rb

```diff
@@ -0,0 +1,314 @@
+require "rails_helper"
+
+describe "Nested documentable" do
+  factories = [
+    :budget_investment,
+    :dashboard_action,
+    :proposal
+  ]
+
+  let(:factory) { factories.sample }
+  let!(:documentable) { create(factory) }
+  let!(:user) { create(:user, :level_two) }
+  let(:path) { attachable_path_for(factory, documentable) }
+  let(:submit_button_text) { submit_button_text_for(factory, path) }
+  let(:notice_text) { notice_text_for(factory, path) }
+
+  context "New and edit path" do
+    before do
+      Setting["uploads.documents.max_amount"] = 1
+      create(:administrator, user: user) if admin_section?(path)
+      documentable.update!(author: user) if edit_path?(path)
+    end
+
+    scenario "Shows or hides new document link depending on max documents limit" do
+      do_login_for(user, management: management_section?(path))
+      visit path
+
+      expect(page).to have_link id: "new_document_link"
```

Comment:
Since we're at it:

```suggestion
      expect(page).to have_link "Add new document"
```

Same for the `not_to have_css` expectation below.